doc/specs/Keyboard-Selection.md

```diff
@@ -0,0 +1,143 @@
+
+By default, the following keybindings will be set:
+```JS
+// Cell Selection
+{ "command": { "action": "moveSelectionAnchor", "direction": "down" },  "keys": ["shift+down"] },
+{ "command": { "action": "moveSelectionAnchor", "direction": "up" },    "keys": ["shift+up"] },
+{ "command": { "action": "moveSelectionAnchor", "direction": "left" },  "keys": ["shift+left"] },
+{ "command": { "action": "moveSelectionAnchor", "direction": "right" }, "keys": ["shift+right"] },
+
+// Word Selection
+{ "command": { "action": "moveSelectionAnchor", "direction": "left",    "expansionMode": "word" }, "keys": ["ctrl+shift+left"] },
+{ "command": { "action": "moveSelectionAnchor", "direction": "right",   "expansionMode": "word" }, "keys": ["ctrl+shift+right"] },
+
+// Viewport Selection
+{ "command": { "action": "moveSelectionAnchor", "direction": "left",    "expansionMode": "viewport" }, "keys": ["shift+home"] },
+{ "command": { "action": "moveSelectionAnchor", "direction": "right",   "expansionMode": "viewport" }, "keys": ["shift+end"] },
+{ "command": { "action": "moveSelectionAnchor", "direction": "up",      "expansionMode": "viewport" }, "keys": ["ctrl+shift+up"] },
+{ "command": { "action": "moveSelectionAnchor", "direction": "down",    "expansionMode": "viewport" }, "keys": ["ctrl+shift+down"] },
+
+// Buffer Corner Selection
+{ "command": { "action": "moveSelectionAnchor", "direction": "up",      "expansionMode": "buffer" }, "keys": ["ctrl+shift+home"] },
+{ "command": { "action": "moveSelectionAnchor", "direction": "down",    "expansionMode": "buffer" }, "keys": ["ctrl+shift+end"] },
+
+// Select All
+{ "command": "selectEntireBuffer", "keys": ["ctrl+a"] },
+
+// Mark Mode
+{ "command": "toggleMarkMode", "keys": ["ctrl+m"] },
+```
+
+## Capabilities
+
+### Accessibility
+
+Being able to create and update a selection without the use of a mouse is a very important feature to users with disabilities. The UI Automation system (particularly with the use of the signaling model), should be aware that a new selection exists. Thus, the bounding rects should be updated and the associated text regions should be reread appropriately.
+
+### Security
+
+N/A
+
+### Reliability
+
+With regards to the Terminal Core, the newly introduced code should rely on already existing and tested code. Thus no crash-related bugs are expected.
+
+With regards to Terminal Control and the settings model, crash-related bugs are not expected. However, ensuring that the selection is updated and cleared in general use-case scenarious must be ensured.
+
+### Compatibility
+
+N/A
+
+### Performance, Power, and Efficiency
+
+## Potential Issues
+
+The settings model makes all of these features easy to disable, if the user wishes to do so. So, no.
+
+## Future considerations
+
+### Expanding Mark Mode functionality
+This spec currently limits mark mode functionality to the following:
+- only move by cell (no other expansion modes)
+- no rebinding arrow keys allowed

```

Comment:
What, why? You listed every other binding above. Why leave these four out? They're only functional after one has entered mark mode in the first place.

Reply:
Yeah. Check out the thread above. This may be worth talking about in person too. :/

Reply:
Yeah maybe on Monday.